### ISSUE TITLE
10000+ Warnings issued

### ISSUE THREAD

**Issue by aliasal** [brython-dev/brython#1167]
Hi
I am new to brython, so I hope this post is at the right place.
I wrote my first script in brython, it works, but there are more than 10000 (no typo ten thousand) warnings issued, which lead to 10seconds passing before the script is finished and the site is loaded.
Of course I thought the problem to be with my code, so I reduced it, till I only had the imports:

from browser import window, document, html, alert, console
from fractions import Fraction
from random import random, sample

This already leads to thousands of warnings. Further reducing leads to the fractions module as the culprit.
As I use Fractions in my code, more and more warnings are issued, which slow down the execution.
Although I don't think my code is the problem, here it is:

[test.zip](https://github.com/brython-dev/brython/files/3405814/test.zip)

**Comment by PierreQuentel** [comment]
Sorry, I can't reproduce the issue. I tried the code you sent with Firefox and Chrome on Windows 10 and got no warning at all. Can you explain which kind of message you get in the console ?

**Comment by aliasal** [comment]
> 
> 
> Sorry, I can't reproduce the issue. I tried the code you sent with Firefox and Chrome on Windows 10 and got no warning at all. Can you explain which kind of message you get in the console ?

Thanks for the fast reply. It is always "the same" warning:

TypeError: can't assign to property "__class__" on 94: not an object 			        brython.js:5697:1
TypeError: can't assign to property "__class__" on -429120515313559: not an object 	brython.js:5697:1
all referring to 5697:1
and one info:  using indexedDB for stdlib modules cache
Just the number that is "not an object" is changing wildly from small to large and negative to positive.

Sorry that I didn't include info about my system: Windows 10, 64bit, Firefox, both up to date
I use brython 3.7.4 which seems to give more warnings than 3.7.3
Chrome seems indeed to issue no warnings (but I never use it, so maybe I am overlooking something) it just nags [Violation] that the success handler took 3351ms.

**Comment by PierreQuentel** [comment]
Thanks for the information, it might be because the JS engine doesn't like setting the attribute `__class__` to integers. Maybe it depends on browser settings.

Can you try with the version of `brython.js` in this commit and tell me if the warnings are still there ?

**Comment by PierreQuentel** [comment]
Thanks for the information, it might be because the JS engine doesn't like setting the attribute `__class__` to integers. Maybe it depends on browser settings.

Can you try with the version of `brython.js` in this commit and tell me if the warnings are still there ?

**Comment by aliasal** [comment]
Thanks a lot. 
That has fixed most of the warnings. "Only" 367 left.
In case you are interested in them, about 200 are:
ReferenceError: assignment to undeclared variable $dec7128  brython.js line 8859 > Function:237:20
The rest is like this (some of them more than once):
SyntaxError: test for equality (==) mistyped as assignment (=)?  brython.js:7412:134
SyntaxError: applying the 'delete' operator to an unqualified name is deprecated  brython.js:9003:7
ReferenceError: reference to undefined property "$type" brython.js:65:4
ReferenceError: assignment to undeclared variable method_descriptor brython.js:5633:4
ReferenceError: reference to undefined property "$nat" brython.js:6186:4
ReferenceError: reference to undefined property "__class__" brython.js:13184:25
ReferenceError: reference to undefined property "C" brython.js:1183:7
ReferenceError: reference to undefined property "ntype" brython.js:3167:4
ReferenceError: reference to undefined property "parent" brython.js:3789:7
ReferenceError: reference to undefined property "br\xFCche" brython.js:420:40
ReferenceError: reference to undefined property "limit_denominator" brython.js:3310:4
TypeError: useless expression brython.js:177:20 
ReferenceError: reference to undefined property "__func__" brython.js:5482:55
ReferenceError: assignment to undeclared variable module brython.js:13219:23
ReferenceError: reference to undefined property "__path__" brython.js:8841:16
ReferenceError: reference to undefined property 0 brython.js:5558:4
ReferenceError: reference to undefined property "__isabstractmethod__" brython.js:5378:29
ReferenceError: assignment to undeclared variable init_subclass brython.js:5398:33
ReferenceError: assignment to undeclared variable extra_kw brython.js:5654:93
TypeError: can't assign to property "$hash" on 0: not an object brython.js:11879:1
TypeError: can't assign to property "$hash" on 1: not an object brython.js:11879:1
TypeError: can't assign to property "$hash" on 2: not an object brython.js:11879:1
TypeError: can't assign to property "$hash" on "_order_": not an object brython.js:11879:1
TypeError: can't assign to property "$hash" on "_ignore_": not an object brython.js:11879:1
TypeError: can't assign to property "$hash" on "sign": not an object brython.js:11879:1
TypeError: can't assign to property "$hash" on "\\.": not an object 2 brython.js:11879:1
TypeError: can't assign to property "$hash" on "int": not an object brython.js:11879:1
TypeError: can't assign to property "$hash" on "\\.": not an object 2 brython.js:11879:1
TypeError: can't assign to property "$hash" on "frac": not an object brython.js:11879:1
TypeError: can't assign to property "$hash" on "exp": not an object brython.js:11879:1
TypeError: can't assign to property "$hash" on "signal": not an object brython.js:11879:1
TypeError: can't assign to property "$hash" on "diag": not an object brython.js:11879:1
TypeError: can't assign to property "$hash" on 80: not an object 2 brython.js:11879:1
TypeError: can't assign to property "$hash" on "iso885910": not an object brython.js:11879:1

Sorry for spamming you with these. 
All warnings are in the attached file.
[warnings.txt](https://github.com/brython-dev/brython/files/3411889/warnings.txt)

**Comment by PierreQuentel** [comment]
These commits should remove some of the warnings, not all (the "reference to undefined property" should remain).
Are there specific browser settings to get these warnings ? If so, and since they slow down the execution, in normal use they should be disabled because the reported issues don't break the code. But I would like to know how to get them for debugging.
Thanks again !

**Comment by aliasal** [comment]
Thanks again. On first load the page was down to 40 warnings, but after reload the count has gone up to about 250 (file attached).
As I am using Firefox for over a decade now, I have totally forgotten about changes I may have made over the years. But I looked it up and indeed, in 
about:config  -> javascript.options.strict
I set the value to true (false is default). 
After setting it back to false, all warnings are gone.
[warnings2.txt](https://github.com/brython-dev/brython/files/3416421/warnings2.txt)
